### DISCUSSION TITLE
Spoliers tags for achievements

### DISCUSSION THREAD

**Discussion by kdecks** [RetroAchievements/RAIntegration#776] [Ideas]
(RA web issue, as this is a hybrid suggestion: https://github.com/RetroAchievements/RAWeb/issues/275)

As a player playing a game for the first time, especially a story driven one, I'd like to have the option for plot elements in achievements and descriptions hidden from me.

This is especially true for RPGs, but can apply to all sorts of games. Reading about characters I will meet, places I will go, bosses I will encounter, all of these could be minor or major spoilers.

Here's what I propose via a series of images: 
(the images are to convey the idea, not the ideal placement or design)

(for RASuite) Allow achievements to be tagged by the developer:
![image](https://user-images.githubusercontent.com/32706333/52157075-8f4a7380-2649-11e9-9590-246e45e12dfe.png)

In the emulator the tagged achievment would be hid:
![image](https://user-images.githubusercontent.com/32706333/52157142-03851700-264a-11e9-8478-744733c26051.png)
Once earned they would be revealed.

A display switch for the emulator:
![image](https://user-images.githubusercontent.com/32706333/52157700-37623b80-264e-11e9-9072-7f19b51a6d96.png)

A setting switch on the game page:
![image](https://user-images.githubusercontent.com/32706333/52157557-1baa6580-264d-11e9-8432-188d67ebeaf0.png)
Once earned they will be revealed.

A global setting switch from My Settings:
![image](https://user-images.githubusercontent.com/32706333/52157518-cc643500-264c-11e9-929c-c5928898b04d.png)

Other stuff/Questions:
- Should spoilers be hidden for users not signed in?
- How else could this be presented?
- About retroarch, it could be left out, or if someone was interested added in as well, but the emulator hide/reveal settings are intended for RA emus.
- Would spoiler achievement also be hidden from the global feed?

**Comment by Jamiras** [comment]
I realize this is an "opt-in" sort of feature, but it reminds me of the unwelcome concept about the player not knowing what they need to do:
> "Secret Achievements", where the players have no indication of what they're going after. If you want to prevent spoilers give at least a clue on the Achievement Title/Description. Otherwise the players will start to search on web or mess with Achievement Editor/Memory Inspector, and in the process they can stumble across even worse spoilers.

**Comment by wilhitewarrior** [comment]
I have to disagree that the unwelcome concept applies to this idea. The player can choose to see it if they want to. If anything this is for the player's protection so they don't accidentally read something they don't want to while looking through the achievement list. If the player has already played the game or they just don't care about spoilers, they'll still have the option to unhide it if they so wish. Besides, this would be more for story-based progression achievements that aren't missable in the first place most of the time.

With the recent support PSX and NDS, consoles that have games with much deeper and richer stories than older games did, avoiding spoilers in achievement descriptions while still describing the objective is becoming more and more difficult, borderline impossible sometimes.

**Comment by Jamiras** [comment]
From http://www.voidcn.com/article/p-gmokauqn-bgh.html:

> Avoid secret achievements. The original goal of secret achievements was to prevent spoilers. The downside is that players can't actively pursue an achievement if they don't know what they're going after. Instead, they turn to one of numerous Web sites to find out what the achievement is, and in doing so can stumble across even worse spoilers. The only style of player who would go after all the secret achievements without spoiling the game for themselves would be the hardcore Explorers or Achievers.
>
> Solution: Consider using vague descriptions that avoid spoilers, but give the player some hint about how to go about earning the achievement.

From http://docs.retroachievements.org/Achievement-Design/#challenges:
> One thing that you have to remember about retro gamers - and the games they play - is that most of us have played these games a million times before. 

This is primarily a server-side feature (see https://github.com/RetroAchievements/RAWeb/issues/275). We need database, API, and site support. The client simply has to have a field to specify that an achievement is a spoiler and send it to the server.
